src/hooks/useSetErcBalance.ts

```diff
@@ -0,0 +1,139 @@
+ *  and verify that the value of the storage slot is the same as the balanceOf call
+ */
+export function useSetErcBalance() {
+  const client = useClient()
+
+  return useMutation({
+    async mutationFn({ erc, address, value }: SetErcBalanceParameters) {
+      try {
+        let slotFound = false
+        let slotGuess = 0n
+
+        while (slotFound !== true) {
+          // if mapping, use keccak256(abi.encode(address(key), uint(slot)));
+          const encodedData = encodeAbiParameters(
+            [
+              { name: 'key', type: 'address' },
+              { name: 'slot', type: 'uint' },
+            ],
+            [address, slotGuess],
+          )
+
+          const oldSlotValue = await client.getStorageAt({
+            address: erc,
+            slot: keccak256(encodedData),
+          })
+
+          // user value might be something that might have collision (like 0)
+          await client.setStorageAt({
+            address: erc,
+            index: keccak256(encodedData),
+            value: pad(toHex(SLOT_VALUE_TO_CHECK)),
+          })
+
+          const newBalance = await client.readContract({
+            abi: balanceOfABI,
+            address: erc,
+            functionName: 'balanceOf',
+            args: [address],
+          })
+
+          const guessIsCorrect = newBalance === BigInt(SLOT_VALUE_TO_CHECK)
+
+          if (guessIsCorrect) {
+            slotFound = true
+            await client.setStorageAt({
+              address: erc,
+              index: keccak256(encodedData),
+              value: pad(toHex(value)),
+            })
+          } else {
+            // check for a rebasing token (stETH)
+            // by setting storage value again with an offset
+            await client.setStorageAt({
+              address: erc,
+              index: keccak256(encodedData),
+              value: pad(toHex(SLOT_VALUE_TO_CHECK + 1n)),
+            })
+            const newBalanceAgain = await client.readContract({
+              abi: balanceOfABI,
+              address: erc,
+              functionName: 'balanceOf',
+              args: [address],
+            })
+
+            // the diff in balanceOf is the offset in value
+            if (newBalanceAgain - newBalance === 1n) {
+              slotFound = true
+              await client.setStorageAt({
+                address: erc,
+                index: keccak256(encodedData),
+                value: pad(toHex(value)),
+              })
+              break
+            }
+
+            // reset storage slot
+            await client.setStorageAt({
+              address: erc,
+              index: keccak256(encodedData),
+              value: oldSlotValue || pad('0x0'),
+            })
+
+            // loop
+            slotGuess++
+            if (slotGuess >= 10n) {
+              throw "couldn't find balancesOf mapping past slot 10"
+            }
+          }
+        }
+      } catch (e) {
+        console.error(e)
+      }
```

Comment:
Would be nice to compose this into a `setContractStorage` module in `src/actions`, that we can eventually reuse for other things. 

```ts
setContractStorage({
  abi: [...],
  functionName: '..',
  args: ...,
  value: ...
})
```

Kinda looks like a viem action!